tests/kafkatest/services/kafka.py

```diff
@@ -199,29 +208,53 @@ def restart_node(self, node, wait_sec=0, clean_shutdown=True):
     def leader(self, topic, partition=0):
         """ Get the leader replica for the given topic and partition.
         """
-        cmd = "/opt/kafka/bin/kafka-run-class.sh kafka.tools.ZooKeeperMainWrapper -server %s " \
-              % self.zk.connect_setting()
-        cmd += "get /brokers/topics/%s/partitions/%d/state" % (topic, partition)
-        self.logger.debug(cmd)
+        topic_partition_data = self.zk_get_data("/brokers/topics/%s/partitions/%d/state" % (topic, partition))
+
+        if topic_partition_data is None:
+            raise Exception("Error finding data for topic %s and partition %d." % (topic, partition))
+        self.logger.info(topic_partition_data)
```

Comment:
With stuff like this, I like to ask whether anyone might ever call this in a pretty tight loop, e.g. in order to wait to meet some condition (like leader failover) but detect it fairly promptly. In those cases, logging in an accessor can be dangerous (huge amount of log output).